### PR TITLE
Implemented: Support to fetch product details (#2ebfxn4)

### PR DESCRIPTION
**Pull request by rvutd**
Defined function to fetch order details using solr-query on the basis of productId

### CODE REVIEW

File: src/product/index.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import api from '../../api'
+import { Product } from '../../types/Product';
```
Comment (ymaheshwari1) [inline src/product/index.ts:2]:
```suggestion
import { Product } from '../../types';
```
Export product type from the types index.ts file

File: src/product/index.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+async function getProductDetails (productId: string): Promise<any> {
```
Comment (ymaheshwari1) [inline src/product/index.ts:4]:
```suggestion
async function getProductDetails (productId: string): Promise<Product> {
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/product/index.ts (hunk 2 above)
Comment (rvutd) [reply src/product/index.ts]: updated

File: src/product/index.ts (hunk 1 above)
Comment (rvutd) [reply src/product/index.ts]: updated

File: src/product/index.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            isVariant: productDetails.isVariant
+          }        
+          return product;
+        } else return null
+      })      
+      return {products: products, totalVirtual: resp.data.grouped.groupId.ngroups, totalVariant: resp.data.grouped.groupId.matches}
+    }
+  } catch (err) {
+    console.error(err)
+    return null
```
Comment (ymaheshwari1) [inline src/product/index.ts:156]: Update the return value

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/product/index.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+      code: 'error',
+      message: 'something went wrong',
+      serverResponse: err
+    }
+  }
+
+  return response
+}
+
+async function getVariant(variantProductIds: Array<string>): Promise<Array<any>> {
```
Comment (ymaheshwari1) [inline src/product/index.ts:169]:
```suggestion
async function getVariant(variantProductIds: Array<string>): Promise<Array<Product>> {
```

File: src/product/index.ts [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      url: 'solr-query',
+      method: 'post',
+      data: query
+    })
+
+    if (resp?.status == 200 && resp.data?.grouped?.groupId?.groups?.length > 0) {
+      const group = resp.data.grouped.groupId.groups[0]
+      const productDetails = group.doclist.docs[0]
+      const imageUrls = productDetails.additionalImageUrls
+      imageUrls.push(productDetails.mainImageUrl)
```
Comment (ymaheshwari1) [inline src/product/index.ts:32]: I think we don't need to add the main image to the `imageUrls` as we have the mainImage available in another property.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/product/index.ts (hunk 5 above)
Comment (rvutd) [reply src/product/index.ts:32]: updated

File: src/product/index.ts (hunk 4 above)
Comment (rvutd) [reply src/product/index.ts]: updated

**Pull request closed without merging**